### DISCUSSION TITLE
deprecation warning in View.h

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3921] [Q&A]
When I use View.h, Xcode gives me a warning:

```c++
/View.h:204:51: warning: declaration is marked with '@deprecated' command but does not have a deprecation attribute [-Wdocumentation-deprecated-sync]
        float focusDistance = 10.0f;        //!< @deprecated use Camera::setFocusDistance() instead
                                                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I tried changing that member declaration to

```c++
float focusDistance [[deprecated("use Camera::setFocusDistance() instead")]] = 10.0f;
        //!< @deprecated use Camera::setFocusDistance() instead
```

but then whenever you copy or move an instance of the `View::DepthOfFieldOptions` structure, you get a deprecation warning.  I could forget about changing View.h and just turn off the warning by adding the compile flag `-Wno-documentation-deprecated-sync`.   But I wonder if anyone can think of a way to make the deprecation warning only show up if one explicitly touches the `focusDistance` member.

**Comment by bejado** [comment]
Yeah, unfortunately I'm not sure there's a good solution here. It's annoying that Clang still generates the warning when copying `DepthOfFieldOptions`. Maybe we shouldn't be using the `@deprecated` tag for fields.

**Reply by jwwalker** [reply]
If you write a copy constructor that doesn't copy that member, then Clang doesn't issue the warning.  But I don't think "deprecated" should necessarily mean "totally nonfunctional".
